File: JAVA/Singlelevel.java
class Base
{
    public int i;
    public int j;

    public Base()   //Default
    {
        System.out.println("Inside Base Constructor");
    }
    
    public void Fun()
    {
        System.out.println("Inside Base Fun()");
    }
    public void Gun()
    {
        System.out.println("Inside Base Gun()");
    }

}
class Derived extends Base
{
    public int x;
    public int y;

    public Derived()
    {
        System.out.println("Inside Derived Constructor");
    }

    public void Sun()
    {
        System.out.println("Inside Derived Sun()");
    }
}





class Singlelevel 
{
    public static void main(String A[])
    {
            Derived dobj = new Derived();

            dobj.Fun();
            dobj.Gun();
            dobj.Sun();
    }
}

File: CPP/Singlelevel.cpp
#include <iostream>
using namespace std;

class Base
{
    public:

        int i, j;

        Base()
        {
        cout<<"Inside Base Constructor \n";
        }

        void Fun()
        {
            cout<<"Inside Base Fun() \n";
        }
        void Gun()
        {
            cout<<"Inside Base Gun() \n";
        }
        ~Base()
        {
        cout<<"Inside Base Destructor \n";
        }
};
class Derived : public Base
{
    public:
       

        int x, y;

        void  Sun()
        {
            cout<<"Inside Base Sun() \n";
        }

     
        Derived()
        {
            cout<<"Inside Derived Constructor \n";
        }
        ~Derived()
        {
            cout<<"Inside Derived Constructor \n";
        }
     
};

int main()
{

        cout<<sizeof(Base)<<"\n";
         cout<<sizeof(Derived)<<"\n";
   
   
        




    return 0;
}
